cogniac/gateway.py

```diff
@@ -35,23 +50,88 @@ class CogniacGateway(object):
     media from another host on the same network.
     """
 
-    def __init__(self, timeout=60, url_prefix=None):
+    ##
+    #  get
+    ##
+    @classmethod
+    @retry(stop_max_attempt_number=8, wait_exponential_multiplier=500, retry_on_exception=server_error)
+    def get(cls,
+            connection=None,
+            gateway_id=None,
+            url_prefix=None):
+        """
+        get single network camera
+        connnection (CogniacConnection): Authenticated CogniacConnection object
+        gateway_id (String): the unique identifier of a Gateway object
+        returns CogniacGateway object
+        """
+        gateway_dict = None
+        if connection and gateway_id:
+            resp = connection._get("/gateways/{}".format(gateway_id))
+            gateway = resp.json()
+        return CogniacGateway(connection=connection, url_prefix=url_prefix, gateway_dict=gateway)
+
+    ##
+    #  get_all
+    ##
+    @classmethod
+    def get_all(cls, connection):
+        """
+        return all CogniacGateway objects belonging to the currently authenticated tenant
+
+        connnection (CogniacConnection):     Authenticated CogniacConnection object
+        """
+        resp = connection._get('/tenants/%s/gateways' % connection.tenant.tenant_id)
```

Comment:
any exception handling required?

Reply:
Other entities' `get_all_` functions also follow this pattern.